SDL3/FAQDevelopment.md

```diff
@@ -10,18 +45,151 @@ The most portable way to include SDL headers is to use angular quotes around the
 
 No, most graphics back ends are not thread-safe, so you should only call SDL video functions from the main thread of your application. [SDL_RunOnMainThread()](SDL_RunOnMainThread) can be used to dispatch code that needs to run on the main thread.
 
-## Can I call SDL event functions from multiple threads?
+## Can I call SDL event functions from multiple threads? <a name="events_on_threads" id="events_on_threads"></a>
 
 The main event handling should be done on the main thread, though you can use [SDL_PushEvent()](SDL_PushEvent) and [SDL_PeepEvents()](SDL_PeepEvents) to interact with the event queue on other threads. Most SDL functions have their thread-safety noted in their documentation.
 
-## Can I use shaders with the SDL 2D render API?
 
-No, there are many different ways to use shaders, and rather than restrict you to a subset of shader functionality, we recommend using a full 3D API like [SDL GPU](CategoryGPU). You can look at how SDL uses the GPU API for 2D rendering if you'd like to expand on that: [SDL_render_gpu.c](https://github.com/libsdl-org/SDL/blob/main/src/render/gpu/SDL_render_gpu.c)
+## Why is my sleep/SDL-Delay function inaccurate? (Why am I getting 61/62/59 fps and not 60?): <a name="inaccurate_sleep" id="inaccurate_sleep"></a>
+
+Take a look at these blogs for an in-depth understanding of the problems of timing:
+ - https://blog.bearcats.nl/accurate-sleep-function/ (Recentlyish Updated)
+ - https://blog.bearcats.nl/perfect-sleep-function/ (Part 2, even better Sleeps!)
+
+In SDL3 you should be using [`SDL_DelayPrecise`](SDL_DelayPrecise)  (which is similar, albeit not currently perfect to the above) or [`SDL_DelayNS`](SDL_DelayNS) (More precise than SDL_Delay)
+
+On a similar note, when doing timing, prefer using [`SDL_GetPerformanceCounter`](SDL_GetPerformanceCounter)  and [`SDL_GetPerformanceFrequency`](SDL_GetPerformanceFrequency) , or [`SDL_GetTicksNS`](SDL_GetTicksNS) , over using [`SDL_GetTicks`](SDL_GetTicks) . Beyond just sleep inaccuracy, milliseconds isn't precise enough to do timing work.
+
+
+## I'm getting an error about SDL_main, WinMain, main, etc: <a name="main" id="main"></a>
+
+> NOTE: This answer is written for brevity, but this is a complex topic! For a full understanding of what's happening, please look over our [main functions README](README-main-functions).
+
+It's _most_ likely you have `#include "SDL3/SDL_main.h"` somewhere in your code, but your `main` function doesn't look like [what SDL expects](SDL_main), although note, you need not name it `SDL_main`.
+
+If you see an error about "a previous definition of `SDL_main`", there's pretty much two possibilities:
+ - You've implemented `main`/`SDL_main` twice in one file yourself. Pretty unlikely, but it can happen!
+ - More likely you have `#define SDL_MAIN_USE_CALLBACKS` somewhere, or have `SDL_MAIN_USE_CALLBACKS` defined through your build system as a switch to the compiler, but you're implementing `main`/`SDL_main` instead of the callbacks.
```

Comment:
```suggestion
 - More likely you have `#define SDL_MAIN_USE_CALLBACKS` somewhere, or have `SDL_MAIN_USE_CALLBACKS` defined through your build system as a switch to the compiler, but you're implementing `main` instead of or in addition to the callbacks.
```